api_v3/lib/syndication/KalturaSyndicationFeedRenderer.php

```diff
@@ -351,6 +395,67 @@ public function getEntriesIds()
 		}
 		return $entries;
 	}
+
+	public function getNextEntryByPlaylistType()
+	{
+		if ($this->dynamicPlaylist)
+		{
+			return $this->getNextEntryViaDynamic();
+		}
+		else
+		{
+			return $this->getNextEntry();
+		}
+	}
+
+	public function getNextEntryViaDynamic()
+	{
+		if(!$this->executed)
+		{
+			$this->entryIdsHandled = array();
+			$this->entriesCurrentPage = array();
+			$this->lastEntryIds = array();
+			$this->lastEntryCreatedAt = 0;
+		}
+
+		++$this->returnedEntriesCount;
+		if ($this->returnedEntriesCount > $this->limit)
+		{
+			return false;
+		}
+
+		if (!$this->entriesCurrentPage)
+		{
+			$this->fetchNextEntriesAccordingToFilter();
+		}
+
+		if (current($this->entriesCurrentPage))
+		{
+			$entry = entryPeer::retrieveByPK(current($this->entriesCurrentPage)->id);
+			if($entry && !array_key_exists($entry->getId(), $this->entryIdsHandled))
```

Comment:
Check what happen in the current implementation when using multi filters and they return the same object twice.
In any case the return should be false only when no more entries